src/main.ts

```diff
@@ -352,34 +354,34 @@ export default class ReadwisePlugin extends Plugin {
     this.registerInterval(this.scheduleInterval);
   }
 
-  refreshBookExport(bookIds?: Array<string>) {
+  async refreshBookExport(bookIds?: Array<string>) {
     bookIds = bookIds || this.settings.booksToRefresh;
     if (!bookIds.length || !this.settings.refreshBooks) {
       return;
     }
     try {
-      fetch(
+      const response = await fetch(
         `${baseURL}/api/refresh_book_export`,
         {
           headers: {...this.getAuthHeaders(), 'Content-Type': 'application/json'},
           method: "POST",
           body: JSON.stringify({exportTarget: 'obsidian', books: bookIds})

```

Comment:
@tyler-dot-earth we received some requests to this endpoint on our backend with the `books` key undefined -- this would have returned a 500

it was at ~12:42AM UTC.

was that just you messing around with stuff, or potentially a new regression on this PR?

Reply:
@TristanH I was working on a last lil bit of optimizatio. Lemme do a bit more testing to verify, will respond back soon 👍

Reply:
@TristanH Pretty sure the 500s were from WIP code that I hadn't committed. That said, here's some updates:

- i added a *HUGE* list of test cases to the original post so it can be thoroughly tested by others
- i added a commit to straighten out sync/refresh behavior better control when books are removed from the list of book IDs to refresh
- i added a commit to restore missing files when "Resync deleted files" is enabled
  - not 100% happy with the solution (find missing books from bookIdsMap that aren't in the vault), but it was the easiest way to achieve this
  - ^ the core of my issue is that the codebase doesn't really make a distinguishment between books to refresh and books which have been deleted
  - ... so now books that have been deleted do NOT get added to the fresh list; refresh list is now just used as the queue for syncs (meaning deleted files won't end up in that queue unless user has `resync deleted files` enabled)

i can attempt to further explain the rationale as needed (or further refine the logic/code), but it's getting late at the moment.

take a look and try the test cases when you have a bit of time (: i'll run through them again after i sleep to double check everything.

Reply:
Noticing clean install sync isn't working as expected; moving this back draft while I fix. Will update soon.